src/mono/wasm/runtime/web-socket.ts

```diff
@@ -26,7 +26,20 @@ let mono_wasm_web_socket_close_warning = false;
 const ws_send_buffer_blocking_threshold = 65536;
 const emptyBuffer = new Uint8Array();
 
+function verifyEnvironment() {
+    if (ENVIRONMENT_IS_SHELL) {
```

Comment:
Shoud we move it after the `typeof globalThis.WebSocket !== "function"` condition?

Reply:
for some random polyfill ? I don't think so.

Reply:
It has complex dynamic behavior and even the nodeJS polyfill doen't work well in all use-cases. I'm just loud and clear this is not supported.